uber/models/attendee.py

```diff
@@ -827,6 +827,26 @@ def full_name(cls):
             [(or_(cls.first_name == None, cls.first_name == ''), 'zzz')],  # noqa: E711
             else_=func.lower(cls.first_name + ' ' + cls.last_name))
 
+    @hybrid_property
```

Comment:
Attendees' relationships to promo code groups are based on two distinct model-relationships: attendees who claim badges are related to the group through their promo code, and the buyer is related to the group through a separate 'buyer' relationship column. This property is used mostly for the attendee search, letting you pull attendees by promo code groups no matter how they're related to that group.

In hindsight, I kind of question my earlier decision. There's no current plans to let attendees buy multiple promo code groups, and we will almost certainly never let you both claim a badge in a group and be the buyer for another group, so I _could_ have just generated an extra promo code and given that to the buyer, thus precluding the need for a lot of the gymnastics I had to do here. On the other other hand, the way I structured it reflects the real-life reality that the buyer of a group is fundamentally different than the group members, so... maybe it's fine.

Reply:
Yeah, this is probably "better" even if the relationship is more complicated.  I can also see people being responsible for multiple promo code groups.  They totally SHOULDN'T but I can definitely see it happening once we start using promo code groups for the plugins (probably in 2020).  A lot of MIVS presenters / panelists / etc are people who are at least already IN another promo code group and sometimes are the people who registered it.  So I can definitely see this extra bit of complexity in the data model being useful.